Automattic/jetpack
# Add TS check to build process

After the introduction of TypeScript, we need to ensure that no PR breaks any types.

#### Changes proposed in this Pull Request:
* Add `build` script to `js-packages/components` 
* Add TS check to build process for `js-packages/components` and `plugins/jetpack` for now.
* Use concurrently for TS checks to avoid it increasing build time

#### Jetpack product discussion
See **Next Steps** and the related discussion in comments in https://github.com/Automattic/jetpack/pull/24294

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Add a TS error on purpose in some TS component in `js-packages`
    * For example use `clasName` instead of `className` in `projects/js-packages/components/components/record-meter-bar/index.tsx`
* Run `jetpack build js-packages/components -v`
* Confirm that you see the TS error introduced above
* Do something similar in `projects/plugins/jetpack/_inc/client/at-a-glance/backup-upgrade/bar-chart.tsx`
* Run `jetpack build plugins/jetpack -v`
* Confirm that you see the TS error
* Remove any introduced TS errors
* Run the above build commands again
* Run `jetpack build --all`
* Confirm that there is no TS error

**Noote:** Initially the build should fail because we have TS errors in master which are being fixed in #24294
So, we need to fix those TS errors and merge into master before this PR can be merged.
